app/src/renderer/features/session/sessionSlice.ts

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from "@reduxjs/toolkit";
+import type { PayloadAction } from "@reduxjs/toolkit";
+import type { RootState } from "../../store";
+
+enum SessionStatus {
+  Unauth,
+  Offline,
+  Auth,
+}
+
+interface AuthData {
+  expiration: string;
```

Comment:
Nit (fine to defer to follow-up work): When we make use of this value, we'll probably need to retype it:


```suggestion
  expiration?: Date;
```

Reply:
Ack.  Nice to know that some problems rhyme between Python and TypeScript!  :-)